### PR TITLE
The porting of LLA Templates

### PR DESCRIPTION
**Pull request by TDcoolguy300**
Ports LLA's template system as a dedicated subsystem, and of course this took some rewriting for configs. Will show some images here in a moment

The Default thing i included, will always generate due to it being the only one
[image](http://puu.sh/mzHvj/31514bfe5d.png?raw)
The Template panel
[image](http://puu.sh/mzHB6/c15d7ea301.png?raw)
Includes admin logging for this stuff.
[image](http://puu.sh/mzHDG/ad5fd86519.png?raw)

### CODE REVIEW

**Comment by TDcoolguy300** [comment]
Whoops, a test file got caught in this.

**Comment by danhibiki** [comment]
:+1:

File: code/modules/templates/verbs.dm [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+	dat += "</span><br><br>"
+
+	if(length(SStemplate.placed_templates))
+		dat += "<table>"
+		dat += "<tr><th>Name</th><th>Position</th>[check_rights(R_FUN) ? "<th>Actions</th>" : ""]"
+
+		for(var/datum/dmm_object_collection/template in SStemplate.placed_templates)
+			dat += "<tr><td>[template.name]</td><td>{[template.location.x], [template.location.y], [template.location.z]}</td>"
+			if(check_rights(R_FUN))
```
Comment (TDcoolguy300) [inline code/modules/templates/verbs.dm:24]: Now, this needs Headmin inputs more specifically, i am not sure what to assign them to, so i defaulted them to +FUN

**Comment by TDcoolguy300** [comment]
This needs some cleanup from some previous experiments, give me a moment

**Comment by TDcoolguy300** [comment]
Updated OP with images

File: _maps/RandomZLevels/templates/spacegen/asteroid.dmm [hunk 2]
```diff
@@ -0,0 +1,28 @@
+"a" = (/turf/space,/area/space)
```
Comment (ghost) [inline _maps/RandomZLevels/templates/spacegen/asteroid.dmm:1]: i suggest putting these under `_maps/templates` instead of RandomZLevels

**Comment by TDcoolguy300** [comment]
Works on a config option, thats what i just had it set to

**Pull request merged**